Gitweb:     http://git.kernel.org/linus/2ae8bb75db1f3de422eb5898f2a063c46c36dba8
Commit:     2ae8bb75db1f3de422eb5898f2a063c46c36dba8
Parent:     338bac527ed0e35b4cb50390972f15d3cbce92ca
Author:     Tejun Heo <tj@kernel.org>
AuthorDate: Mon Oct 26 15:41:46 2009 +0100
Committer:  Ingo Molnar <mingo@elte.hu>
CommitDate: Sun Nov 8 13:19:05 2009 +0100

    x86: Fix iommu=nodac parameter handling
    
    iommu=nodac should forbid dac instead of enabling it. Fix it.
    
    Signed-off-by: Tejun Heo <tj@kernel.org>
    Acked-by: FUJITA Tomonori <fujita.tomonori@lab.ntt.co.jp>
    Cc: Matteo Frigo <athena@fftw.org>
    Cc: <stable@kernel.org> # .32.x and older
    LKML-Reference: <4AE5B52A.4050408@kernel.org>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 arch/x86/kernel/pci-dma.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

--- a/arch/x86/kernel/pci-dma.c
+++ b/arch/x86/kernel/pci-dma.c
@@ -214,7 +214,7 @@ static __init int iommu_setup(char *p)
 		if (!strncmp(p, "allowdac", 8))
 			forbid_dac = 0;
 		if (!strncmp(p, "nodac", 5))
-			forbid_dac = -1;
+			forbid_dac = 1;
 		if (!strncmp(p, "usedac", 6)) {
 			forbid_dac = -1;
 			return 1;
